elytron/WFLY-14735-Management-FORM-authentication.adoc

```diff
@@ -0,0 +1,245 @@
+password in addition to the regular password.  A new evidence type will be defined within WildFly Elytron
+which will accept an arbitrary set of attributes during authentication.  The existing security realms
+within WildFly Elytron that handle username/password based authentication will be updated to support
+this new dynamic evidence type.
+
+==== core-management
+
+The number of attributes related to security on the management interfaces has grown with different
+permutations needed for different situations.  To reduce the complexity a new resource `policy=security`
+will be added to the `core-management` subsystem, this will allow references to the sasl or http 
+authentication factories as well as to ssl-contexts to be used.  This new policy resouce will also be
+able to reference the new context made available from the `elytron-authentication` subsystem.
+
+==== Management Interfaces
+
+On the management interfaces a new attribute `use-core-management-security-policy` will be added,
+this will be a boolean and when set to `true` the configuration defined on the `policy=security`
+resource will be used instead of the individual attributes on the management interface.
+
+==== Out Of Scope
+
+The RFE will not be making changes to the SASL authentication for native access, the out of the
+box configuration will still default to supporting the JBOSS_LOCAL_USER and DIGEST SASL mechanisms.
+
+This RFE will also not be changing the getting started experience adding the initial users to the
+server using the add-user utility and this RFE will not be changing the properties files used to
+contain the initial users.
+
+=== Nice-to-Have Requirements
+
+Should the management interfaces have TLS enabled by default, we may be able to also generate a 
+wildfly-config.xml for clients to use poining at a Keystore for the server's certificate.
+
+This could however be high risk breaking various clients.
+
+=== Non-Requirements
+
+The architecture chosen for this RFE will form the basis for future development which will not be covered by
+this RFE.
+
+It is envisaged that the form authentication provided by the `elytron-authentication` subsystem will be 
+enhanced further with capabilities such as:
+
+ * Support for MFA authentication using OTP such as HOTP / TOTP.
+ * Dynamic decisions based on username such as recognising a user is registered for SSO.
+ * Fido2 / WebAuth authentication with or without a username.
+ * Multi factor authentication using push notifications.
+ * Customisation of the look and feel of the form.
+ 
+In addition to using the form for the management interfaces it will be possible to use configurations
+to secure deployments making all of these options available.
+
+These additional features are broadly the reason for placing the initial configuration for the form 
+into a new subsystem.
+
+== Backwards Compatibility
+
+=== Default Configuration
```

Comment:
Regarding the `security` command, will just `*-http-auth-management` be affected?